lib/ansible/modules/network/fortimanager/fmgr_secprof_proxy.py

```diff
@@ -0,0 +1,481 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: fmgr_secprof_proxy
+version_added: "2.8"
+author:
+    - Luke Weighall (@lweighall)
+    - Andrew Welsh (@Ghilli3)
+    - Jim Huber (@p4r4n0y1ng)
+short_description: Manage proxy security profiles in FortiManager
+description:
+  -  Manage proxy security profiles for FortiGates via FortiManager using the FMG API with playbooks
+
+options:
+  adom:
+    description:
+      - The ADOM the configuration should belong to.
+    required: false
+    default: root
+
+  host:
+    description:
+      - The FortiManager's Address.
+    required: true
+
+  username:
+    description:
+      - The username associated with the account.
+    required: true
+
+  password:
+    description:
+      - The password associated with the username account.
+    required: true
+
+  mode:
+    description:
+      - Sets one of three modes for managing the object.
+      - Allows use of soft-adds instead of overwriting existing values
+    choices: ['add', 'set', 'delete', 'update']
+    required: false
+    default: add
+
+  strip_encoding:
+    description:
+      - Enable/disable stripping unsupported encoding from the request header.
+      - choice | disable | Disable stripping of unsupported encoding from the request header.
+      - choice | enable | Enable stripping of unsupported encoding from the request header.
+    required: false
+    choices: ["disable", "enable"]
+
+  name:
+    description:
+      - Profile name.
+    required: false
+
+  log_header_change:
+    description:
+      - Enable/disable logging HTTP header changes.
+      - choice | disable | Disable Enable/disable logging HTTP header changes.
+      - choice | enable | Enable Enable/disable logging HTTP header changes.
+    required: false
+    choices: ["disable", "enable"]
+
+  header_x_forwarded_for:
+    description:
+      - Action to take on the HTTP x-forwarded-for header in forwarded requests| forwards (pass), adds, or removes the
+      -  HTTP header.
+      - choice | pass | Forward the same HTTP header.
+      - choice | add | Add the HTTP header.
+      - choice | remove | Remove the HTTP header.
+    required: false
+    choices: ["pass", "add", "remove"]
+
+  header_x_authenticated_user:
+    description:
+      - Action to take on the HTTP x-authenticated-user header in forwarded requests| forwards (pass), adds, or remove
+      - s the HTTP header.
+      - choice | pass | Forward the same HTTP header.
+      - choice | add | Add the HTTP header.
+      - choice | remove | Remove the HTTP header.
+    required: false
+    choices: ["pass", "add", "remove"]
+
+  header_x_authenticated_groups:
+    description:
+      - Action to take on the HTTP x-authenticated-groups header in forwarded requests| forwards (pass), adds, or remo
+      - ves the HTTP header.
```

Comment:
truncated word. not sure how this will render in documentation. might be worth just moving the full word at the end of 111 to 112...